sunxspex/integrate.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from scipy.integrate._quadrature import _cached_roots_legendre
+from scipy.special import lpmv
+
+
+def _legendre_roots(a, b, npoints, eps=3e-14):
```

Comment:
Is there a reason to keep all this? points and weights can be obtained straight from numpy and the interval transformed from [-1,1] to [a,b] easily, eg:

```
xi,wi=np.polynomial.legendre.leggauss(N)
xm=0.5*(b+a)
xl=0.5*(b-a)
new_xi=np.repeat(xm,n).reshape(xi.shape[0],n)+np.outer(xl,xi)
new_wi=np.outer(wi,xl).T
```

Reply:
I guess my plan was to remove the `gauss_legendre` method and only use the `fixed_quad` one if the results are the same which they seem to be or maybe keep it for backward compatibility and use `fixed_quad` as default